pg_shard.c

```diff
@@ -92,9 +95,11 @@ static Oid ExtractFirstDistributedTableId(Query *query);
 static bool ExtractRangeTableEntryWalker(Node *node, List **rangeTableList);
 static List * DistributedQueryShardList(Query *query);
 static bool SelectFromMultipleShards(Query *query, List *queryShardList);
+static Query * BuildLocalQuery(Query *query, List *localRestrictList);
```

Comment:
I've used a heuristic of trying to maintain the order based on order of the call to the function. So in this case RowAndColumnFilterQuery is being called before BuildLocalQuery. Could you maintain that heuristic (and also make sure the forward declaration order is the same as the implementation order?